docs/specs/sms_gateway.md

```diff
@@ -0,0 +1,73 @@
+# SMS Gateway
+
+This document describes configuration of sms gateway.
+
+- [Using Custom SMS Gateway](#using-custom-sms-gateway)
+  - [Configuration](#configuration)
+  - [Webhook Signature](#webhook-signature)
+  - [Webhook Body](#webhook-body)
+  - [Deno Script](#deno-script)
+
+## Using Custom SMS Gateway
+
+When using custom sms gateway, `messaging.sms_provider` must be set to value `custom` in configuration. A secret with key `sms.custom` must also exist in secret configuration.
+
+### Configuration
+
+The following section must be included in `authgear.yaml`:
+
+```
+messaging:
+  sms_provider: custom
+```
+
+The following section must be included in `authgear.secret.yaml`:
+
+```yaml
+secrets:
+  - data:
+      url: "https://myapp.com/send_sms"
+      secret: "somesecretkey"
+    key: sms.custom
+```
+
+When using a deno script, secret can be unspecified:
+
+```yaml
+secrets:
+  - data:
+      url: "authgeardeno:///deno/randomstring.ts"
+    key: sms.custom
+```
+
+### Webhook Signature
+
+Each request is signed with the secret key provided in the `sms.custom` section with `secret` key. The developer must validate the signature and reject requests with invalid signature to ensure the request originates from Authgear.
+
+The signature is calculated as the hex encoded value of HMAC-SHA256 of the request body.
```

Comment:
Reference the existing definition of webhook signature, instead of repeating it here.

Reply:
@louischan-oursky 🤔 But I think they are actually different things, and actually we are using a different secret to sign the request of sms gateway requests. I think in future the interface can become different too.